sqlmesh/core/context_diff.py

```diff
@@ -183,10 +186,18 @@ def create_no_diff(cls, environment: str) -> ContextDiff:
         Returns:
             The ContextDiff object.
         """
+        env = state_reader.get_environment(environment.lower())
```

Comment:
[Nit] I'd probably just pass the environment here instead of the entire state reader.

Reply:
Also makes it easier to test

Reply:
That's a good point, thank you